doc/pit/Platform_Integration_Test_high_level_design.md

```diff
@@ -0,0 +1,164 @@
+
+### 2.2.1 Development test
+
+This happens on the ODM vendor side, product development stage. PIT can be used to:
+
+  1) Run as daily regression test to check if any new feature breaks the other feature.
+  2) Run in a stress test for any given function.
+  3) Run in environment reliability test, check if environment changes (such as high/low temperature, humidity changes) affect device functions.
+
+### 2.2.2 Manufacturing test
+
+This happens on the ODM vendor side, product manufacturing stage. PIT can be used to:
+
+  1) Run in the process of manufacturing, PIT test can automatically test the devices, with exactly the test logic from user requirement, avoiding mis-understanding between users' requirement and ODM implementation.
+  2) Provide users' minimum test coverage.
+
+### 2.2.3 Delivery test
+
+This happens on the user side, product delivery stage. PIT can be used to:
+
+  1) Provide a fast and automated test for the product, checking the basic functionality of the device hardware, firmware and drivers.
+  2) Provide a structured data as the result of the test, and it can be easily checked by the users' acceptance checking system.
+
+# 3. PIT system design
+
+## 3.1 System view
+
+PIT is supposed to run in a docker, as a component of SONiC. It can be enable/disable in the build process. Also, it's a docker, so it can be deploy/upgrade flexibly on demands.
+
+## 3.2 Objective
+
+* PIT should run in different scenario, with simple user input.
+* PIT should use exactly the same test logic for different product.
+* PIT should generate standard outputs for specific test item.
+* PIT should be extensible.
+
+## 3.3 PIT software architecture
+
+Figure 3-1 shows the high-level software architecture of PIT.
+![](images/PIT_software_architecture.png)
+Figure 3-1
+
+PIT main process parses and interpret user input command and options, invoke test case list generator to generate test case list. The later reads platform configuration and test case congiuration, traverse test case set and selects those cases with desired tags, then generates final list of test cases. Then main process iterates each case selected, run it, then invoke test result generator to generate test result. After all test cases in the list finished, result generator gother them and form a collective final result. The result is passed to users' automation system for acceptance test.
+
+Besides auto generated test cases list, PIT also support single case or user specific case list as input.
+
+## 3.4 PIT component detailed
+Figure 3-2 shows the softawre component of PIT.
+![](images/PIT_conponents.png)
+
+### 3.4.1 Test cases list generator
+
+Test case list generator is used to generate a list of test cases, it parses users' input options, product specific platform configuration, and generate test case list from all test case database. This is used in the daily regression test and delivery test scenario.
```

Comment:
is it possible to reuse the current test framework in the [sonic-mgmt repositoty](https://github.com/Azure/sonic-mgmt)? If the new cases can be added there it will also benefit those who use the sonic-mgmt test cases as SONiC daily regression.

Reply:
I agree with Kebo. It's much easier to put the tests together. 

I'm not following on why standalone test is difficult.

Reply:
> I agree with Kebo. It's much easier to put the tests together.
> 
> I'm not following on why standalone test is difficult.

The most important reason is ODM vendor cannot have access to SAI unless they pay extra money while we(users) have it, especially for new ASIC. In this situation, they cannot have the whole sonic running, this is a reality for most smaller ODM vendors.